File: Go/HBD_Debian.go
package main

import (
	"fmt"
	"math"
)

func main() {
	// --- 1. Set Constants ---
	// Define the dimensions and mathematical constants for the spiral.
	const WIDTH = 40
	const HEIGHT = 22
	const SPIRAL_CONST_A = 1.2

	// These values are calculated at runtime, so they must be variables (`var`), not constants (`const`).
	var GOLDEN_RATIO = (1 + math.Sqrt(5)) / 2
	var SPIRAL_CONST_B = math.Log(GOLDEN_RATIO) / (math.Pi / 2)


	// --- 2. Define a Canvas ---
	// Create a 2D slice of runes (Go's equivalent for characters)
	// and initialize it with space characters.
	canvas := make([][]rune, HEIGHT)
	for i := range canvas {
		canvas[i] = make([]rune, WIDTH)
		for j := range canvas[i] {
			canvas[i][j] = ' '
		}
	}

	// --- 3. Loop through the Angle (theta) ---
	// Go uses a `for` loop to replicate the behavior of a `while` loop.
	theta := 0.0
	for theta < 6*math.Pi {
		// Calculate the polar coordinates of the spiral.
		radius := SPIRAL_CONST_A * math.Exp(SPIRAL_CONST_B*theta)
		x := radius * math.Cos(theta)
		y := -radius * math.Sin(theta)

		// Convert spiral coordinates to screen coordinates.
		// We must explicitly cast the float64 results to int for slice indexing.
		screenX := int(x*2 + WIDTH/2)
		screenY := int(y + HEIGHT/2)

		// Check if the coordinates are within the canvas bounds before drawing.
		if screenX >= 0 && screenX < WIDTH && screenY >= 0 && screenY < HEIGHT {
			canvas[screenY][screenX] = '$'
		}

		// Increment the angle.
		theta += 0.02
	}

	// Place the '@' symbol at the center of the spiral.
	canvas[HEIGHT/2][WIDTH/2] = '@'

	// --- 7. Print the Canvas ---
	fmt.Println("\n--- Happy Birthday Debian!! ---")
	// To match the Python output, we iterate through the rows in reverse order (from HEIGHT-1 down to 0).
	for i := HEIGHT - 1; i >= 0; i-- {
		row := canvas[i]
		reversedRow := make([]rune, WIDTH)
		// We also reverse the characters in each row (from WIDTH-1 down to 0).
		for j := WIDTH - 1; j >= 0; j-- {
			reversedRow[WIDTH-1-j] = row[j]
		}
		// Print the fully reversed row as a string.
		fmt.Println(string(reversedRow))
	}
	fmt.Println("\n         Debian                  ")
	fmt.Println("------------------------------------------- \n")
}

File: Rust/HBD_Debian.rs
use std::f64::consts::PI;

fn main() {
    // --- 1. Set Constants & Variables ---
    // Define the dimensions of the canvas. These can be `const` as they are literals.
    const WIDTH: usize = 40;
    const HEIGHT: usize = 22;
    const SPIRAL_CONST_A: f64 = 1.2;

    // These values must be calculated at runtime, so we use `let` instead of `const`.
    // The `sqrt` and `ln` functions are not `const fn` and cannot be evaluated at compile time.
    let golden_ratio: f64 = (1.0 + 5.0_f64.sqrt()) / 2.0;
    let spiral_const_b: f64 = golden_ratio.ln() / (PI / 2.0);

    // --- 2. Define a Canvas ---
    // Create a 2D vector (a vector of vectors) to act as the character grid.
    // It's initialized with space characters, just like the Python version.
    let mut canvas = vec![vec![' '; WIDTH]; HEIGHT];

    // --- 3. Loop through the Angle (theta) ---
    // Loop from 0 up to 6*PI to draw a multi-turn spiral.
    let mut theta = 0.0;
    while theta < 6.0 * PI {
        // Calculate the radius for the current angle using the logarithmic spiral formula.
        let radius = SPIRAL_CONST_A * (spiral_const_b * theta).exp();

        // Convert polar coordinates (radius, theta) to Cartesian coordinates (x, y).
        let x = radius * theta.cos();
        // The y-axis is inverted to match typical screen coordinate systems (y increases downwards).
        let y = -radius * theta.sin();

        // Scale and translate the Cartesian coordinates to fit onto the canvas.
        // We cast to `isize` first to handle potential negative coordinates before the bounds check.
        let screen_x = (x * 2.0 + (WIDTH as f64) / 2.0) as isize;
        let screen_y = (y + (HEIGHT as f64) / 2.0) as isize;

        // --- 4. Draw on Canvas ---
        // Check if the calculated screen coordinates are within the canvas bounds.
        if screen_x >= 0 && screen_x < WIDTH as isize && screen_y >= 0 && screen_y < HEIGHT as isize
        {
            // If they are, place the '$' character. We must cast back to `usize` for indexing.
            canvas[screen_y as usize][screen_x as usize] = '$';
        }

        // Increment the angle for the next point.
        theta += 0.02;
    }

    // Place the '@' character at the center of the spiral.
    canvas[HEIGHT / 2][WIDTH / 2] = '@';

    // --- 5. Print the Canvas ---
    println!("\n--- Happy Birthday Debian!! ---");

    // Iterate through the canvas's rows in reverse order to print from bottom to top.
    for row in canvas.iter().rev() {
        // For each row, create a new reversed string and print it.
        // This correctly orients the spiral as it was in the original script.
        let reversed_row: String = row.iter().rev().collect();
        println!("{}", reversed_row);
    }

    println!("\n         Debian              ");
    println!("------------------------------------------- \n");
}
